core/src/main/java/org/testcontainers/utility/AuditLogger.java

```diff
@@ -0,0 +1,67 @@
+package org.testcontainers.utility;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.dockerjava.api.command.DockerCmd;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.MDC;
+
+import java.util.List;
+
+/**
+ * Logger for tracking potentially destructive actions, intended for usage in a shared Docker environment where
+ * traceability is needed. This class uses SLF4J, logging at TRACE level and capturing common fields as MDC fields.
+ * <p>
+ * Users should configure their test logging to apply appropriate filters/storage so that these logs are
+ * captured appropriately.
+ */
+@Slf4j
+@UtilityClass
+public class AuditLogger {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    public static final String MDC_PREFIX = AuditLogger.class.getCanonicalName();
+
+    public static void doLog(String action, String image, String containerId, DockerCmd<?> cmd) {
+        doLog(action, image, containerId, cmd, null);
+    }
+
+    public static void doLog(String action, String image, String containerId, DockerCmd<?> cmd, @Nullable Exception e) {
+        MDC.put(MDC_PREFIX + ".Action", action);
+        MDC.put(MDC_PREFIX + ".Image", image);
+        MDC.put(MDC_PREFIX + ".ContainerId", containerId);
+        try {
+            MDC.put(MDC_PREFIX + ".Command", objectMapper.writeValueAsString(cmd));
+        } catch (JsonProcessingException ignored) {
+        }
+
+        if (e != null) {
+            MDC.put(MDC_PREFIX + ".Exception", e.getLocalizedMessage());
+            log.trace("{} action with image: {}, containerId: {}", action, image, containerId, e);
```

Comment:
Everything is logged at TRACE level, so it will be generally invisible for most users. People who want to use this will have to set up an SLF4J log level override.

Also note use of MDC; this is to allow key-value pairs to be associated with each event. We use this at $WORK to avoid regex/grokking of log message strings.